HW6: Chausov Andrey

dxahtepb opened the pull request.

dxahtepb committed:
[task-6] Implement Pair<T, K> interface

Walingar commented on src/main/java/impl/pair/NumberPairFactory.java:
```diff
@@ -4,6 +4,6 @@
 public class NumberPairFactory {
     public static <K extends Number, T extends Number> Pair<K, T> of(K first, T second) {
-        return null;
+        return new PairImpl<>(first, second);
```
Тут по условию задания нужно было вернуть реализацию NumberPair

dxahtepb committed:
[task 6] Code review fixes

- Return NumberPair implementation from NumberPairFactory
- Use instanceof in PairImpl::equals()
